datanode/src/storage_api.py

```diff
@@ -292,17 +297,86 @@ def GridCellsOperationDataHandler(zoom, gufi):
   return _FormatResult(result)
 
 
+@webapp.route(
+  '/UVR/<zoom>/<message_id>',
+  methods=['PUT', 'POST', 'DELETE'])
+def UvrDataHandler(zoom, message_id):
+  """Handles the web service request for emplacing or removing a UVR.
+
+  Args:
+    zoom: zoom level in slippy tile format
+    message_id: unique message ID of constraint
+    OAuth access_token as part of the header
+  Returns:
+    200 with token and metadata in JSON format,
+    or the nominal 4xx error codes as necessary.
+  """
+  uss_id = _ValidateAccessToken(UVR_WRITE_SCOPE)
+
+  if not uss_id:
+    return {
+      'status': 'fail',
+      'code': status.HTTP_400_BAD_REQUEST,
+      'message': 'uss_id must be provided in the request to modify a UVR.'}
+
+  uvr_json = _GetRequestParameter('uvr', None)
+  if not uvr_json:
+    return {
+      'status': 'error',
+      'code': status.HTTP_400_BAD_REQUEST,
+      'message': 'uvr field must be provided in the form data request'}
+
+  result = {}
+  try:
+    unvalidated_uvr = json.loads(uvr_json)
+  except ValueError as e:
+    abort(status.HTTP_400_BAD_REQUEST, 'Error parsing UVR JSON: ' + e.message)
+
+  try:
+    uvr = uvrs.Uvr(unvalidated_uvr)
+
+    if uvr['message_id'] != message_id:
+      raise ValueError('message_id "%s" in request does not match message_id '
+                       '"%s" in UVR' % (message_id, uvr['message_id']))
+    if uvr['originator_id'] != uss_id:
```

Comment:
uss_id is created by the OAuth provider, originator_id is created by the USS, and I don't think we want the USS to specify it. In gridcell metadata, we specify a request object that does not contain the fields we write (version, uss_id, timestamp, etc.).

Reply:
I think we had previously discussed that a USS, even one with UVR write scope, should not be allowed to delete a UVR emplaced by by a different USS.  If originator_id is not uss_id, does that mean we also need an additional ```emplaced_by_uss_id``` field to be stored in the grid cell?

Reply:
I was saying originator_id is uss_id, and the USS does not specify it (it comes from their OAuth token).